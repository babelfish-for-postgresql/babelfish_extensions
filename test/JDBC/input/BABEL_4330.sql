
-- We should be able to use replace function in computed column.
CREATE TABLE babel_4330_t1(a varchar(50) NULL, b as replace(a, '1', '2'));
GO

INSERT INTO babel_4330_t1 VALUES('13131');
GO

SELECT * FROM babel_4330_t1
GO

UPDATE babel_4330_t1 SET a = '14141' WHERE a = '13131';
GO

SELECT * FROM babel_4330_t1
GO

ALTER TABLE babel_4330_t1 ADD c AS replace(a, '1','5');
GO

SELECT * FROM babel_4330_t1
GO


CREATE TABLE babel_4330_t2(a varchar(50) NULL, b as replace(a, NULL, '2'));
GO

INSERT INTO babel_4330_t2 VALUES('13131');
GO

SELECT * FROM babel_4330_t2
GO

CREATE TABLE babel_4330_t3(a varchar(50) NULL, b as replace(a, '1', NULL));
GO

INSERT INTO babel_4330_t3 VALUES('13131');
GO

SELECT * FROM babel_4330_t3
GO

DROP TABLE babel_4330_t1;
GO


DROP TABLE babel_4330_t2;
GO


DROP TABLE babel_4330_t3;
GO

