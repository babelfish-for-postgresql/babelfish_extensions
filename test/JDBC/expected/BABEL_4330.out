
-- We should be able to use replace function in computed column.
CREATE TABLE babel_4330_t1(a varchar(50) NULL, b as replace(a, '1', '2'));
GO

INSERT INTO babel_4330_t1 VALUES('13131');
GO
~~ROW COUNT: 1~~


SELECT * FROM babel_4330_t1
GO
~~START~~
varchar#!#text
13131#!#23232
~~END~~


UPDATE babel_4330_t1 SET a = '14141' WHERE a = '13131';
GO
~~ROW COUNT: 1~~


SELECT * FROM babel_4330_t1
GO
~~START~~
varchar#!#text
14141#!#24242
~~END~~


ALTER TABLE babel_4330_t1 ADD c AS replace(a, '1','5');
GO

SELECT * FROM babel_4330_t1
GO
~~START~~
varchar#!#text#!#text
14141#!#24242#!#54545
~~END~~



CREATE TABLE babel_4330_t2(a varchar(50) NULL, b as replace(a, NULL, '2'));
GO

INSERT INTO babel_4330_t2 VALUES('13131');
GO
~~ROW COUNT: 1~~


SELECT * FROM babel_4330_t2
GO
~~START~~
varchar#!#text
13131#!#<NULL>
~~END~~


CREATE TABLE babel_4330_t3(a varchar(50) NULL, b as replace(a, '1', NULL));
GO

INSERT INTO babel_4330_t3 VALUES('13131');
GO
~~ROW COUNT: 1~~


SELECT * FROM babel_4330_t3
GO
~~START~~
varchar#!#text
13131#!#<NULL>
~~END~~


DROP TABLE babel_4330_t1;
GO


DROP TABLE babel_4330_t2;
GO


DROP TABLE babel_4330_t3;
GO

